### DISCUSSION TITLE
Standalone parser documentation

### DISCUSSION THREAD

**Discussion by gl-yziquel** [PaddiM8/kalk#125] [Ideas]
Hi.

I'd like to use the `kalk` parsing library to perform various custom stuff, among which some nonlinear and semidefinite programming. As such, I have little use for the `kalker` tool, but would appreciate using the `kalk` parsing library more thoroughly. Which brings me to three points:

1. As of today, it seems to me that the parsing library is tightly coupled with the evaluation code. Ideally, that should be separate: I want to do my own evaluation and syntactic manipulation.

2. I do not see documentation for the purely kalk parsing features. I would like to declare some function, and have it parsed, and recovered in my code as an AST. Documentation for that is lacking.

3. Would it be conceivable to migrate the parsing technology to some parser combinator library like `nom` to allow code reuse ? I'd like to reuse the parsing technology to include some stuff from abstract differential geometry on top of `kalk`. `nom` would allow for the kind of code reuse I would need.

Is attending to these two issues potentially on the roadmap for kalk ?

**Comment by PaddiM8** [comment]
Hmm I am not sure about exposing those parts. Also, migrating the parsing to something like nom would likely not be possible since it has context dependencies, unfortunately.

**Reply by gl-yziquel** [reply]
Not sure what you mean by context dependencies, but I managed to toy around with some nom code that handles contextual information while parsing using the flat_map combinator. It's much more usable in v8.0.0-alpha2, now.

**Comment by gl-yziquel** [comment]
Then I'll have to fork it.
